### PR TITLE
[Projects] Change the background task timeout for project deletion

### PR DESCRIPTION
**Pull request by moranbental**
When creating the background task for project deletion, the timeout should be configured using the default value `mlrun.mlconf.background_tasks.default_timeouts.operations.delete_project` from `config.py`, which is set to 15 minutes. However, the background task was hardcoded to wait only 6 minutes. 

The fix is to use the default value from the config to ensure consistency
https://iguazio.atlassian.net/browse/ML-7686
https://iguazio.atlassian.net/browse/ML-6595

### CODE REVIEW

**Review by quaark — APPROVED** [review]
👑

File: server/api/utils/clients/iguazio.py [hunk 1]
```diff
@@ -610,7 +618,7 @@ def _verify_job_in_terminal_state():
 
         job_state, job_result = mlrun.utils.helpers.retry_until_successful(
             self._wait_for_job_completion_retry_interval,
-            360,
+            timeout,
```
Comment (alonmr) [inline server/api/utils/clients/iguazio.py]: config timeout is a string
```suggestion
            int(timeout),
```

**Review by alonmr — COMMENTED** [review]
Looks great! minor comment

**Review by alonmr — APPROVED** [review]

File: server/api/utils/clients/iguazio.py [hunk 2]
```diff
@@ -290,7 +290,11 @@ def delete_project(
                 name=name,
                 job_id=job_id,
             )
-            self._wait_for_job_completion(session, job_id)
+            self._wait_for_job_completion(
+                session,
+                job_id,
+                mlrun.mlconf.background_tasks.default_timeouts.operations.delete_project,
```
Comment (alonmr) [inline server/api/utils/clients/iguazio.py]: Actually
```suggestion
                timeout=mlrun.mlconf.background_tasks.default_timeouts.operations.delete_project,
```

**Review by alonmr — CHANGES_REQUESTED** [review]
I think you passed the timeout as error message

**Review by alonmr — APPROVED** [review]

**Review by liranbg — APPROVED** [review]

**Pull request merged**